### ISSUE TITLE
ActiveMQ broker connection doesn't survive broker restarts

### ISSUE THREAD

**Issue by astrobokonon** [LowellObservatory/DataServants#9]
Subscribe to topics, get data, restart broker underneath, and ensure that data still flows when the broker comes back up.  Not currently working reliably, so may need to explicitly check subscription status and/or check for heartbeats now

**Comment by astrobokonon** [comment]
I think I tracked the issue down to an exception that gets thrown in a different thread when attempting `self.conn.start()` in my `amqHelper` class.  It's frustrating, since `self.conn` is of type `stomp.connect.StompConnection11` and the `.start()` method is depreciated and does nothing, but if I omit it, it *never* connects for some reason.  Even though an exception is caught since `stomp.connect.StompConnection11.connect()` raises it in the current thread.

**Comment by astrobokonon** [comment]
When I remove `self.conn.start()` it completely fails to connect, and always throws a `stomp.exception.NotConnectedException` on every attempt. 

BUT I think I'm an idiot; directly inspecting `stomp.Connection` from VSCode, `stop.connect.BaseConnection.start()` **isn't** depreciated, so I think I have a version vs. documentation problem.  Updating STOMP.py and trying again.

**Comment by astrobokonon** [comment]
Note: 
```python 
stomp.__version__ == (4, 1, 19)
```

From their changelog:
```
Version 4.1.20 - Feb 2018

 * Tidy up API (remove need for start/stop methods on the connection)
 * Make updating listeners thread-safe (https://github.com/jasonrbriggs/stomp.py/pull/174)
 * Merge patch from Scott W for configuring grace period for heartbeat timeouts (https://github.com/jasonrbriggs/stomp.py/pull/180)
 * Fix ack/nack in CLI
 * Add missing on_receiver_loop_completed handler to ConnectionListener
```

**Comment by astrobokonon** [comment]
So the whole problem was caused by me not setting the `first` flag in the main Iago code to `False` after the first run was completed.  This caused the check for reconnection to always be skipped, because it was looking at

```python
if conn.conn is None:
    print("No connection at all! Retrying...")
    conn.connect()
elif conn.conn.transport.connected is False and first is False:
    # Added the "first" flag to take care of a double sub. bug
    print("Connection died! Reestablishing...")
    conn.connect()
else:
    print("Connection still valid")
```

The whole STOMP.py version thing and `.start()` depreciation was a red herring.  The underlying ConnectionRefusedError is still being thrown and uncatchable in the thread that STOMP spawns for it, but at least I can reconnect to the broker after the sleep interval is up and the connection is rechecked.

**Comment by astrobokonon** [comment]
![dondraperdrunk](https://user-images.githubusercontent.com/1168555/40023701-cd66a852-5780-11e8-98fa-8c6a05f581d0.gif)

Closed by 788000f93b01c8f2efb86e66a483696d3e8ebdbb
